Sources/SwiftFormat/API/Configuration.swift

```diff
@@ -66,6 +67,9 @@ public struct Configuration: Codable, Equatable {
   /// The maximum length of a line of source code, after which the formatter will break lines.
   public var lineLength: Int
 
+  // Number of spaces that preceeds line comments.
+  public var spacesBeforeLineComments: Int
```

Comment:
Does this apply to both `//` and `/**/` comments? The test case with `/**/` has 1-space in both the input and expected, so wasn't clear from there.

Reply:
I believe `/* */` is referred to as "block comments" regardless of whether it's on the same line or not. In existing unit tests (adjacent to the one I added), there are `/* */` comments with 1 leading space that aren't affected by the formatting (which ensures 2 spaces). I made sure in the documentation, it explicitly says "comments starting with '//'" to avoid this ambiguity.

Reply:
Ah I see, thanks. Could you make the comment here a doc comment (`///`) and include that here as well? @allevato is it intended that block comments are treated differently in this case? Not sure I have much of an opinion on it either way.

Reply:
The name of this configuration should be explicit that it's just about "**end of line** comments", since that traditionally means `//` comments that specifically follow something else on a line and does not include `/* ... */`, `///`, or `/** ... */` comments, nor `//` comments on their own line. So I'd call this `spacesBeforeEndOfLineComments`.

I think it's correct to only support this for `//` EoL comments. Let's imagine we have a situation where we want 2 spaces before those, but only one space around inline block comments, and we have this contrived example:

```swift
let x = 5 // something
let y = someValue + x /* meters */ / someOtherValue /* seconds */
```
When we add another space before the comment on the first line, I don't think it would be right to change the spacing around the block comments on the second line.